Workday/canvas-kit
# docs: Modal example a11y polish

## Summary

Fixes: none 



While revising the accessibility guidelines for the Modal component, I was examining the component examples and came up with a few polishing enhancements for accessibility: 
1. Custom Focus: Set the description of the modal dialog container using `aria-describedby` to the body text message. 
2. Full Overflow: Set `tabindex` on the overflow scrollable area for keyboard support
3. Without Close Icon: Reverse the order of the Delete and Cancel buttons so the least destructive action is first; use ref to set focus to the Cancel button; set the description of the Modal dialog container using `aria-describedby` to the body text. 

I reversed the order of the Delete and Cancel buttons in the WithoutCloseIcon storybook example, so the least destructive action will receive keyboard focus first when modal appears. 


## Release Category
Documentation

---

## Checklist

- [x] MDX documentation adheres to Canvas Kit's [Documentation Guidelines](https://workday.github.io/canvas-kit/?path=/docs/guides-documentation-guidelines--page)

## For the Reviewer




## Where Should the Reviewer Start?



`/modules/react/modal/stories/examples/CustomFocus.tsx`
`/modules/react/modal/stories/examples/FullOverflow.tsx`
`/modules/react/modal/stories/examples/WithoutCloseIcon.tsx`

## Testing Manually



CustomFocus: 
1. Turn on VoiceOver; Launch Safari, and open CustomFocus storybook example, 
2. Tab focus and activate "Acknowledge License" button, 
3. Verify modal appears, 
5. Validate: does VoiceOver announce body text "Enter your initials to acknowledge the license" automatically?
6. Inspect: accessibility tree must show a `description` on the `dialog` container equal to the body text. 

FullOverflow:
1. Open FullOverflow storybook example, 
2. Tab focus the scrollable area, 
3. Validate Up / Down arrow scroll the text into view without a mouse. 

WithoutCloseIcon:
1. Turn on VoiceOver; Launch Safari, and open WithoutCloseIcon storybook example, 
2. Tab focus and activate "Delete" button, 
3. Verify modal appears, 
5. Validate focus is set to "Cancel" button, 
6. Validate: does VoiceOver announce body text "Are you sure you want to delete the item?" automatically?
7. Inspect: accessibility tree must show a `description` on the `dialog` container equal to the body text.
